odl/util/utility.py

```diff
@@ -223,31 +223,55 @@ def __new__(cls, name, this_bases, d):
     return metaclass('temporary_class', None, {})
 
 
+def cache_arguments(function):
+    """Decorate function to cache the result with given arguments.
+
+    This is equivalent to `functools.lru_cache` with Python 3, and currently
+    does nothing with Python 2 but this may change at some later point.
```

Comment:
I'll leave it for now. Python 2 is mostly supported because we have to in a sense. Minor optimizations of 10% should not really be expected on legacy platforms.